server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -149,4 +150,14 @@ private TransferFileSnapshot prepareMetadata(TransferSnapshot transferSnapshot)
 
         return fileSnapshot;
     }
+
+    public void deleteTranslog(long primaryTerm, long generation) throws IOException {
+        String ckpFileName = Translog.getCommitCheckpointFileName(generation);
+        String translogFilename = Translog.getFilename(generation);
+        // ToDo - Take care of metadata file cleanup
```

Comment:
Created a backlog issue : https://github.com/opensearch-project/OpenSearch/issues/5677

Reply:
Do we really need to be tracking checkpoint files or only translog files, since we only care about the latest checkpoint?